modules/standard/Types.chpl

```diff
@@ -711,4 +711,110 @@ inline proc _bor_id(type t) return 0:t;
 pragma "no doc"
 inline proc _bxor_id(type t) return 0:t;
 
+
+/* Return the number of fields in a class or record as a param.
+   The count of fields includes types and param fields.
+ */
+proc numFields(type t) param
+  return __primitive("num fields", t);
+
+/* Get the name of the `i`th field in a class or record.
+   `i` should be in 1..numFields(t).
+
+   :arg t: a class or record type
+   :arg i: which field to get the name of
+   :returns: the name of the field, as a param string
+ */
+proc getFieldName(type t, param i:int) param
+  return __primitive("field num to name", t, i);
+
```

Comment:
If i is not in range, what happens?  If something bad, can we make sure i is in 1..numFields here in module code?  If something reasonable, can you add documentation for it?

Related: "should" feels a little passive-aggressive if in fact it must be in that range.